app/upgrades.go

```diff
@@ -1,3 +1,160 @@
+			}
+
+			app.Logger().Info("move all community pool funds from x/distribution to x/community")
+			FundCommunityPoolModule(ctx, app.distrKeeper, app.bankKeeper, app)
+
+			app.Logger().Info("granting x/gov module account x/community module authz messages")
+			GrantGovCommunityPoolMessages(ctx, app.authzKeeper, app.accountKeeper)
+
+			return toVM, nil
+		},
+	)
+
+	upgradeInfo, err := app.upgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(err)
+	}
+
+	// note: no store updates
+	doUpgrade := upgradeInfo.Name == MainnetUpgradeName || upgradeInfo.Name == TestnetUpgradeName
+	if doUpgrade && !app.upgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		storeUpgrades := storetypes.StoreUpgrades{}
+
+		// configure store loader that checks if version == upgradeHeight and applies store upgrades
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
+	}
+}
+
+// GrantGovCommunityPoolMessages grants x/gov module account access to submit x/authz messages from the community pool module account.
+func GrantGovCommunityPoolMessages(
+	ctx sdk.Context,
+	authzKeeper authzkeeper.Keeper,
+	accountKeeper authkeeper.AccountKeeper,
+) {
+	communityAddr := accountKeeper.GetModuleAddress(communitytypes.ModuleName)
+	govAddr := accountKeeper.GetModuleAddress(govtypes.ModuleName)
+	allowedMsgs := getCommunityPoolAllowedMsgs()
+	for _, msg := range allowedMsgs {
+		auth := authz.NewGenericAuthorization(msg)
+		if err := authzKeeper.SaveGrant(ctx, govAddr, communityAddr, auth, nil); err != nil {
+			panic(fmt.Errorf("failed to grant msg %s to x/gov account: %w", msg, err))
+		}
+	}
+}
+
+// MoveCommunityPoolFunds takes the full balance of the original community pool (the auth fee pool)
+// and transfers them to the new community pool (the x/community module account)
+func FundCommunityPoolModule(
+	ctx sdk.Context,
+	distKeeper distrkeeper.Keeper,
+	bankKeeper bankkeeper.Keeper,
+	app App,
+) {
+	// get balance of original community pool
+	balance, leftoverDust := distKeeper.GetFeePoolCommunityCoins(ctx).TruncateDecimal()
+	app.Logger().Info(fmt.Sprintf("community pool balance: %v, dust: %v", balance, leftoverDust))
+
+	// the balance of the community fee pool is held by the distribution module.
+	// transfer whole pool balance from distribution module to new community pool module account
+	err := bankKeeper.SendCoinsFromModuleToModule(
+		ctx,
+		distrtypes.ModuleName,
+		communitytypes.ModuleAccountName,
+		balance,
+	)
+	if err != nil {
+		panic(fmt.Errorf(
+			"failed to transfer community pool funds to new community pool module account: %w",
+			err,
+		))
+	}
+
+	// make sure x/distribution knows that there're no funds in the community pool.
+	// we keep the leftover decimal change in the account to ensure all funds are accounted for.
+	feePool := distKeeper.GetFeePool(ctx)
+	feePool.CommunityPool = leftoverDust
+	distKeeper.SetFeePool(ctx, feePool)
+}
+
+func getCommunityPoolAllowedMsgs() []string {
```

Comment:
this could just be a variable, though a func works. but i would make it public so it can be imported into e2e tests

```go
var V22UpgradeCommunityModuleGrants = []string{}
```